src/gmp/models/entitymodel.ts

```diff
@@ -25,49 +27,67 @@ import {map} from 'gmp/utils/array';
-  name?: string;
-  end_time?: string;
-  timestamp?: string;
-  permissions?: {
-    permission: EntityModelPermissionElement | EntityModelPermissionElement[];
-  };
-  user_tags?: {
-    tag: UserTagElement | UserTagElement[];
-  };
-  writable?: YesNo;
-  orphan?: YesNo;
-  active?: YesNo;
-  trash?: YesNo;
-  in_use?: YesNo;
-  owner?: Owner;
-  summary?: string;
-  comment?: string;
-}
-
-export interface EntityModelProperties extends BaseModelProperties {
-  active?: YesNo;
-  comment?: string;
-  endTime?: GmpDate;
-  inUse?: boolean;
-  name?: string;
-  orphan?: YesNo;
-  owner?: Owner;
-  summary?: string;
-  timestamp?: GmpDate;
-  trash?: YesNo;
-  userCapabilities?: Capabilities;
-  userTags?: UserTag[];
-  writable?: YesNo;
-}
+export const ownerSchema = z.object({
+  name: z.string(),
+});
+
+export type Owner = z.infer<typeof ownerSchema>;
+
+export const EntityModelPermissionElementSchema = z.object({
+  name: z.string(),
+});
+
+export type EntityModelPermissionElement = z.infer<
+  typeof EntityModelPermissionElementSchema
+>;
+
+export const EntityModelElementSchema = BaseModelElementSchema.extend({
+  name: z.string().optional(),
+  end_time: z.string().optional(),
+  timestamp: z.string().optional(),
+  permissions: z
+    .object({
+      permission: z.union([
+        EntityModelPermissionElementSchema,
+        z.array(EntityModelPermissionElementSchema),
+      ]),
+    })
+    .optional(),
+  user_tags: z
+    .object({
+      tag: z.union([UserTagElementSchema, z.array(UserTagElementSchema)]),
+    })
+    .optional(),
+  writable: z.custom<YesNo>().optional(),
+  orphan: z.custom<YesNo>().optional(),
+  active: z.custom<YesNo>().optional(),
+  trash: z.custom<YesNo>().optional(),
+  in_use: z.custom<YesNo>().optional(),
+  owner: ownerSchema.optional(),
+  summary: z.string().optional(),
+  comment: z.string().optional(),
+});
+
+export type EntityModelElement = z.infer<typeof EntityModelElementSchema>;
+
+export const EntityModelPropertiesSchema = BaseModelPropertiesSchema.extend({
+  active: z.union([z.literal(YES_VALUE), z.literal(NO_VALUE)]).optional(),
+  creationTime: z.instanceof(dayjs as unknown as typeof Dayjs).optional(),
+  endTime: z.instanceof(dayjs as unknown as typeof Dayjs).optional(),
+  comment: z.string().optional(),
+  inUse: z.boolean().optional(),
+  name: z.string().optional(),
+  orphan: z.union([z.literal(YES_VALUE), z.literal(NO_VALUE)]).optional(),
+  owner: ownerSchema.optional(),
+  summary: z.string().optional(),
+  timestamp: z.instanceof(dayjs as unknown as typeof Dayjs).optional(),
+  trash: z.union([z.literal(YES_VALUE), z.literal(NO_VALUE)]).optional(),
+  userCapabilities: z.instanceof(Capabilities).optional(),
+  userTags: z.array(z.instanceof(UserTag)).optional(),
+  writable: z.union([z.literal(YES_VALUE), z.literal(NO_VALUE)]).optional(),
+});
+
+export type EntityModelProperties = z.infer<typeof EntityModelPropertiesSchema>;
```

Comment:
validation should not be necessary for XYZProperties. That's what's going out of the parsing code. Issues with that types should be caught with TypeScript already.